### PR TITLE
etcd-tester: close gRPC connection when canceling

### PR DESCRIPTION
**Pull request by gyuho**
Currently gRPC connection just gets recreated
for every Stress call. When Stress ends or gets
canceled, gRPC connection must also be closed.

For https://github.com/coreos/etcd/issues/4464.

/cc @xiang90

### CODE REVIEW

**Comment by xiang90** [comment]
@gyuho This is racy. Can we add locking properly for stresser?

**Comment by gyuho** [comment]
You are right. Will add locks. Thanks.

**Comment by gyuho** [comment]
@xiang90 PTAL. Added locks.
As you suggest, I will try to use wait group for better synchronizations.

File: tools/functional-tester/etcd-tester/stresser.go [hunk 1]
```diff
@@ -64,6 +65,9 @@ func (s *stresser) Stress() error {
 	if err != nil {
 		return fmt.Errorf("%v (%s)", err, s.Endpoint)
 	}
+	s.mu.Lock()
+	s.conn = conn
+	s.mu.Unlock()
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/stresser.go]: check line 74. s.cancel also need to be protected.

File: tools/functional-tester/etcd-tester/stresser.go [hunk 2]
```diff
@@ -98,6 +102,9 @@ func (s *stresser) Stress() error {
 
 func (s *stresser) Cancel() {
 	s.cancel()
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/stresser.go]: s.cancel need to be protected.

Comment (gyuho) [inline tools/functional-tester/etcd-tester/stresser.go]: Ok

**Comment by gyuho** [comment]
@xiang90 Just added locks around `cancel` member as well. PTAL. Thanks.

File: tools/functional-tester/etcd-tester/stresser.go [hunk 3]
```diff
@@ -97,7 +102,10 @@ func (s *stresser) Stress() error {
 }
 
 func (s *stresser) Cancel() {
+	s.mu.Lock()
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/stresser.go]:
```go
s.mu.Lock()
cancel, conn := s.cancel, s.conn
s.mu.Unlock()
cancel()
conn.Close()
```

File: tools/functional-tester/etcd-tester/stresser.go [hunk 4]
```diff
@@ -97,7 +102,11 @@ func (s *stresser) Stress() error {
 }
 
 func (s *stresser) Cancel() {
-	s.cancel()
+	s.mu.Lock()
+	cancel, conn := s.cancel, s.conn
+	s.mu.Unlock()
+	cancel()
+	conn.Close()
```
Comment (gyuho) [inline tools/functional-tester/etcd-tester/stresser.go]: @xiang90 Just fixed as you suggest. Thanks.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**